File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0;
        int n = nums.size();

        for (int i = 0; i < n; ++i) {
            int total = 0;
            for (int j = i; j < n; ++j) {
                total += nums[j];
                if (total == k) {
                    count++;
                }
            }
        }

        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix_sum_count = defaultdict(int)
        prefix_sum_count[0] = 1  # Handle case where subarray starts from index 0

        current_sum = 0
        count = 0

        for num in nums:
            current_sum += num
            count += prefix_sum_count[current_sum - k]
            prefix_sum_count[current_sum] += 1

        return count